numpy/lib/stride_tricks.py

```diff
@@ -113,6 +113,101 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def sliding_window_view(x, shape, subok=False, writeable=False):
+    """
+    Creates sliding window views of the N dimensional array with the given window
+    shape. Window slides across each dimension of the array and extract subsets of
+    the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create sliding window views from.
+
+    shape : sequence of int
+        The shape of the window. Must have same length as the number of input array dimensions.
+
+    subok : bool, optional
+        If True, then sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool
+        When true, allow writing to the returned view. The default is false, as this should be used
+        with caution: the returned view contains the same memory location multiple times, so
+        writing to one location will cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window views of the array. (view.shape[i] = x.shape[i] - shape[i] + 1)
+
+    See also
+    --------
+    as_strided: Create a view into the array with the given shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    For some cases there may be more efficient approaches to calculate transformations
+    across multi-dimensional arrays, for instance `scipy.signal.fftconvolve`, where combining
+    the iterating step with the calculation itself while storing partial results can result
+    in significant speedups.
+
+    Examples
+    --------
+    >>> i, j = np.ogrid[:3,:4]
```

Comment:
Would be nice to have a more understandable example. Say a 1-D one to start with. Also, to create a 2-D array it's better to use a more regularly used function, e.g. `np.arange(12).reshape((3, 4))`

Reply:
I think the current example is a nice one, but agreed we should start with a 1d example